common/src/main/java/com/wynntils/screens/maps/GuildMapScreen.java

```diff
@@ -55,7 +62,14 @@ public final class GuildMapScreen extends AbstractMapScreen {
     private BasicTexturedButton territoryDefenseFilterButton;
     private BasicTexturedButton hybridModeButton;
 
-    private GuildMapScreen() {}
+    private GuildMapScreen() {
+        // When outside of the main map, center to the middle of the map
+        if (!MathUtils.isInside(
+                (int) McUtils.player().getX(), (int) McUtils.player().getZ(), MIN_X, MAX_X, MIN_Z, MAX_Z)) {
+            updateMapCenter(MAP_CENTER_X, MAP_CENTER_Z);
+            setZoom(0);
+        }
```

Comment:
I don't think this should be decided in the constructor, since some screens hold references to screen instances long before they are opened. Create a `firstInit` boolean, and run this there.